src/Microsoft.AspNetCore.Routing/Matchers/Candidate.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal readonly struct Candidate
+    {
+        public Candidate(Endpoint endpoint, MatchProcessor[] segments)
+        {
+            Endpoint = endpoint;
```

Comment:
`Debug.Assert` not null these arguments?

Reply:
I'm not of fan of using `Debug.Assert` where something trivial is going on. I think there's a really small risk of this regressing.